### ISSUE TITLE
🐛 [firebase_auth] Cannot set tenantId back to null after setting it to something else

### ISSUE THREAD

**Issue by Rexios80** [FirebaseExtended/flutterfire#7896]
## Bug report

The platform code does not allow setting the tenantId to null. Even though it defaults to null, setting it back to null after setting it to something else does not work. (There are null checks in the platform code. It's pretty obvious why it isn't working.)

### Steps to reproduce

1. Set tenantId
2. Do some auth method to see that the tenantId is set
3. Attempt to set tenantId to null
4. Do some auth method to see that the tenantId is still not null

### Expected behavior

Setting the tenantId to null actually sets it to null in the platform code

### Sample project

Working on a PR

**Comment by Salakar** [comment]
Blocked on https://github.com/firebase/firebase-android-sdk/issues/3398

**Comment by stee1ix** [comment]
I am facing this issue as well. Any solutions to this or workarounds?

**Comment by Lyokone** [comment]
Hi 👋 Thanks for the detailed report.

After investigation, this isn't something we can fix in FlutterFire — the root cause is in the underlying native Firebase/Flutter SDK, and FlutterFire only passes the call through here. The fix has to land upstream, where it's already tracked:

https://github.com/firebase/firebase-android-sdk/issues/3398

To avoid a duplicate that we can't action, we're closing this on the FlutterFire side and following the upstream issue instead — once a new SDK version ships with the fix, it'll flow into FlutterFire automatically. Please subscribe to the link above for progress.

If you think there's something FlutterFire-specific we could do in the meantime (a guard, a workaround, a better error), let us know with a minimal repro and we'll happily reopen. Thanks for your patience!
